spec.bs

```diff
@@ -8643,48 +8671,44 @@ To <dfn>build trusted key value scoring signals request body</dfn> given a non-e
   1. Let |mapKey| be a tuple of (|request|'s [=trusted scoring signals request/owner origin=],
     |request|'s [=trusted scoring signals request/joining origin=]).
   1. If |compressionIdMap| does not [=map/contain=] |mapKey|:
-    1. Set |compressionIdMap|[|mapKey|] to |nextCompressionGroupId|.
+    1. [=map/Set=] |compressionIdMap|[|mapKey|] to |nextCompressionGroupId|.
     1. Increment |nextCompressionGroupId| by 1.
   1. Let |compressionGroupId| be |compressionIdMap|[|mapKey|].
-  1. If |compressionGroupMap| does not [=map/contain=] |compressionGroupId|:
-    1. Set |compressionGroupMap|[|compressionGroupId|] to an empty [=map=], whose [=map/keys=] are integers
-  1. Let |partitionId| be [=list/size=] of |compressionGroupMap|[|compressionGroupId|].
-  1. Set |renderUrlIdMap|[|request|'s [=trusted scoring signals request/render URL=]] to [=tuple=]
+  1. If |compressionGroupMap| does not [=map/contain=] |compressionGroupId|, then set
+    |compressionGroupMap|[|compressionGroupId|] to an empty [=map=], whose [=map/keys=] are integers.
+  1. Let |partitionId| be [=map/size=] of |compressionGroupMap|[|compressionGroupId|].
+  1. [=map/Set=] |renderUrlIdMap|[|request|'s [=trusted scoring signals request/render URL=]] to [=tuple=]
     of (|compressionGroupId|, |partitionId|).
-  1. Let |partition| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
-      [=list=] of [=strings=].
-  1. Set |partition|["id"] to |partitionId|.
-  1. Let |namespace| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
-      [=list=] of [=strings=].
-  1. Set |namespace|["render_url"] to [|request|'s [=trusted scoring signals request/render URL=]].
-  1. Set |namespace|["ad_component_render_urls"] to |request|'s [=trusted scoring signals request/ad
-    component URLs=].
-  1. Set |partition|["namespace"] to |namespace|.
-  1. Let |partitionMetadata| be an empty [=map=], whose [=map/keys=] and [=map/values=] are [=strings=].
-  1. Set |partitionMetadata|["experiment_group_id"] to |firstRequest|'s [=trusted scoring signals
-      request/seller experiment group id=].
-  1. Set |partition|["metadata"] to |partitionMetadata|.
-  1. Set |compressionGroupMap|[|compressionGroupId|][|partitionId|] to |partition|.
+  1. [=map/Set=] |compressionGroupMap|[|compressionGroupId|][|partitionId|] to a [=scoring partition=] with the
+    following [=struct/items=]:
+    : [=scoring partition/id=]
+    :: |partitionId|
+
+    : [=scoring partition/namespace=]
+    :: "render_urls" → [|request|'s [=trusted scoring signals request/render URL=]]
+    :: "ad_component_render_urls" → [=trusted scoring signals request/ad component URLs=]

```

Comment:
this is not quite right. If you feel the literal is too big to inline, you can pull it out to declare a map first for it and then just use that variable here.
```suggestion
    :: The [=ordered map=] «[ "render_urls" → « |request|'s [=trusted scoring signals request/render URL=] »,
      "ad_component_render_urls" → |request|'s  [=trusted scoring signals request/ad component URLs=] ]»
```